lib/Driver/DarwinToolChains.cpp

```diff
@@ -427,6 +427,22 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
     }
   }
     
+  if (job.getKind() == LinkKind::Executable) {
```

Comment:
Er, you don't need this part anymore, right? And you never wanted it to just be Executables-linked-with-swiftc anyway.

Reply:
We still ultimately want this to only be linked into executables. We were going to handle that as a followup. For these specific backfill entry points, they're small enough that burning them into dylibs and having potentially-mismatching versions in different images is probably OK, but that's not generally a situation we want to be in.

Reply:
I don't understand that part. We can't rely on executables having *any* Swift in them, and it's unlikely that Xcode will switch to linking with swiftc anyway. The autolinking part for dynamic replacement isn't going away.

Reply:
(I do want a resolution for this because it's interfering with me rebasing #24787.)

Reply:
If there's any Swift in the process, then the executable is the only useful place to put backward-compatibility fills, so much like arclite, we might want eventually for Xcode to link this in unconditionally because a non-Swift executable could be using dylibs with Swift in them.